src/ServiceControlInstaller.Packaging.UnitTests/DeploymentPackageTests.cs

```diff
@@ -0,0 +1,34 @@
+namespace Tests
+{
+    using System.Linq;
+    using NUnit.Framework;
+
+    [TestFixtureSource(typeof(DeploymentPackage), "All")]
+    public class DeploymentPackageTests
+    {
+        DeploymentPackage deploymentPackage;
+
+        public DeploymentPackageTests(DeploymentPackage deploymentPackage)
+        {
+            this.deploymentPackage = deploymentPackage;
+        }
+
+        [Test]
+        public void DuplicateAssemblyFileSizesShouldMatch()
+        {
+            using (var zip = deploymentPackage.Open())
+            {
+                var mainEntries = zip.Entries.Where(x => x.FullName.StartsWith(deploymentPackage.ServiceName)).ToList();
+
+                CollectionAssert.IsNotEmpty(mainEntries, $"Expected a {deploymentPackage.ServiceName} folder in {deploymentPackage.FullName}");
+
+                CollectionAssert.IsEmpty(
+                    from mainEntry in mainEntries
+                    join entry in zip.Entries.Except(mainEntries) on mainEntry.Name equals entry.Name
+                    where entry.Length != mainEntry.Length
+                    select entry.FullName, 
+                    $"File sizes should match the ones in the {deploymentPackage.ServiceName} folder. Check versions of dependencies.");
```

Comment:
Is it sizes or file names?

Reply:
The file names match. We're doing the join on file name. The sizes do not match.

So if you have `Newtonsoft.Json.dll` in the `ServiceControl` folder and in the `Transports/SqlServer` folder but the file sizes don't match, then we have a problem the files are obviously different. One of them will overwrite the other at install time.